gnomad/utils/vcf.py

```diff
@@ -92,3 +339,488 @@ def get_pipe_expr(array_expr: hl.expr.ArrayExpression) -> hl.expr.StringExpressi
+    """
+    edges_dict = {
+        f"{prefix}het": "|".join(
+            map(lambda x: f"{x:.1f}", ht.head(1).age_hist_het.collect()[0].bin_edges)
+        ),
+        f"{prefix}hom": "|".join(
+            map(lambda x: f"{x:.1f}", ht.head(1).age_hist_hom.collect()[0].bin_edges)
+        ),
+    }
+    for hist in HISTS:
+
+        # Parse hists calculated on both raw and adj-filtered data
+        for hist_type in ["adj_qual_hists", "qual_hists"]:
+            hist_name = hist
+            if "adj" in hist_type:
+                hist_name = f"{hist}_adj"
+            edges_dict[hist_name] = (
+                "|".join(
+                    map(
+                        lambda x: f"{x:.2f}",
+                        ht.head(1)[hist_type][hist].collect()[0].bin_edges,
+                    )
+                )
+                if "ab" in hist
+                else "|".join(
+                    map(
+                        lambda x: str(int(x)),
+                        ht.head(1)[hist_type][hist].collect()[0].bin_edges,
+                    )
+                )
+            )
+    return edges_dict
+
+
+def make_hist_dict(bin_edges: Dict[str, Dict[str, str]], adj: bool) -> Dict[str, str]:
+    """
+    Generate dictionary of Number and Description attributes to be used in the VCF header, specifically for histogram annotations.
+
+    :param bin_edges: Dictionary keyed by histogram annotation name, with corresponding string-reformatted bin edges for values.
+    :param adj: Whether to create a header dict for raw or adj quality histograms.
+    :return: Dictionary keyed by VCF INFO annotations, where values are Dictionaries of Number and Description attributes.
+    """
+    header_hist_dict = {}
+    for hist in HISTS:
+
+        # Get hists for both raw and adj data
+        if adj:
+            hist = f"{hist}_adj"
+
+        edges = bin_edges[hist]
+        hist_fields = hist.split("_")
+        hist_text = hist_fields[0].upper()
+
+        if hist_fields[2] == "alt":
+            hist_text = hist_text + " in heterozygous individuals"
+        if adj:
+            hist_text = hist_text + " calculated on high quality genotypes"
+
+        hist_dict = {
+            f"{hist}_bin_freq": {
+                "Number": "A",
+                "Description": f"Histogram for {hist_text}; bin edges are: {edges}",
+            },
+            f"{hist}_n_smaller": {
+                "Number": "A",
+                "Description": f"Count of {hist_fields[0].upper()} values falling below lowest histogram bin edge {hist_text}",
+            },
+            f"{hist}_n_larger": {
+                "Number": "A",
+                "Description": f"Count of {hist_fields[0].upper()} values falling above highest histogram bin edge {hist_text}",
+            },
+        }
+
+        header_hist_dict.update(hist_dict)
+
+    return header_hist_dict
+
+
+def sample_sum_check(
```

Comment:
This is something we want to check on the internal HT sanity checks right? I think it should also be moved somewhere else like the other function I mentioned above, but again not positive where to put it